src/cosim/utility/concurrency.hpp

```diff
@@ -28,75 +28,6 @@ namespace cosim
 namespace utility
 {
 
-
-/**
- *  A thread-safe, fiber-friendly, single-item container.
- *
- *  This is a general-purpose container that may contain zero or one item(s)
- *  of type `T`.
- *
- *  The `put()` and `take()` functions can be safely called from different
- *  threads.  Shared data is protected with synchronization primitives
- *  from Boost.Fiber, meaning that the functions will yield to other fibers,
- *  and not block the current thread, if they cannot immediately acquire a
- *  lock.
- */
-template<typename T>
-class shared_box
-{
-public:
-    using value_type = T;
-
-    /// Puts an item in the container, replacing any existing item.
-    void put(const value_type& value)
-    {
-        {
-            std::lock_guard<boost::fibers::mutex> lock(mutex_);
-            value_ = value;
-        }
-        condition_.notify_one();
-    }
-
-    /// Puts an item in the container, replacing any existing item.
-    void put(value_type&& value)
-    {
-        {
-            std::lock_guard<boost::fibers::mutex> lock(mutex_);
-            value_ = std::move(value);
-        }
-        condition_.notify_one();
-    }
-
-    /**
-     *  Removes an item from the container and returns it.
-     *
-     *  If there is no item in the container when the function is called,
-     *  the current fiber will yield, and it will only resume when an
-     *  item becomes available.
-     */
-    value_type take()
-    {
-        std::unique_lock<boost::fibers::mutex> lock(mutex_);
-        condition_.wait(lock, [&] { return value_.has_value(); });
-        auto value = std::move(value_.value());
-        value_.reset();
-        return value;
-    }
-
-    /// Returns `true` if there is no item in the container.
-    bool empty() const noexcept
-    {
-        std::lock_guard<boost::fibers::mutex> lock(mutex_);
-        return !value_.has_value();
-    }
-
-private:
-    std::optional<value_type> value_;
-    mutable boost::fibers::mutex mutex_;
-    boost::fibers::condition_variable condition_;
-};
-
-
 /**
  *  A shared mutex à la `std::shared_mutex`, but with support for fibers.
```

Comment:
I don't think we need this class at all anymore, we can just use `std::shared_mutex`.

Reply:
Nevermind, I see that @ljamt already suggested we do this in a separate PR.

Reply:
Yes, we noticed this as well. In addition `cosim::utility::shared_mutex` was only used in `utility_concurrency_unittest.cpp`, with the intention to test that file locking functioned correctly with the custom mutex, which was in turn only necessary because of fibers. 

With the removal of fibers, that test can be removed as well. We wanted to push this as a separate PR to avoid more noise in this one, but if we all agree they can be removed, might as well do it here.